src/vos/vos_pool.c

```diff
@@ -979,10 +983,12 @@ vos_pool_query(daos_handle_t poh, vos_pool_info_t *pinfo)
 	pinfo->pif_gc_stat = pool->vp_gc_stat;
 
 	cpi = umem_off2ptr(&pool->vp_umm, pool_df->pd_chk);
-	if (cpi != NULL)
+	if (cpi != NULL) {
 		pinfo->pif_chk = *cpi;
-	else
+		D_ASSERT(pool->vp_feats & VOS_POOL_FEAT_CHK);
```

Comment:
A check like this should prevent setting pd_chk in the first place but in my grepping, I haven't been able to find anywhere it's actually set.

Reply:
This part is reserved for DAOS check pass7. But in this milestone, we only implement pass0-pass5. So there is no real use cases in current code.